torchmultimodal/diffusion_labs/models/adm_unet/adm.py

```diff
@@ -378,7 +346,65 @@ def forward(
             if hidden_states:
                 h = torch.cat([h, hidden_states.pop()], dim=1)
             h = block(h, res_conditional_embedding, attn_conditional_embedding)
-        return h
+
+        # If model is predicting variance, then it should be configured to output double the channels as input
```

Comment:
This should probably be broken out as it's own method (_get_variance_value) to clean up the forward method a bit. If predict_variance_value is false it can return None. Then you can just return DiffusionOutput with both prediction and variance_value as the default variance_value is None anyway.